.github/workflows/vm-kernel.yaml

```diff
@@ -153,46 +201,76 @@ jobs:
           registry: cache.neon.build
           username: ${{ secrets.NEON_CI_DOCKERCACHE_USERNAME }}
           password: ${{ secrets.NEON_CI_DOCKERCACHE_PASSWORD }}
-
-      - name: get kernel version
-        id: get-kernel-version
+      - name: Process kernel image tags
+        id: generate-full-image-tags
         run: |
-          linux_config=$(ls neonvm-kernel/linux-config-*)  # returns something like "neonvm-kernel/linux-config-amd64-6.1.63"
-          kernel_version=${linux_config##*-}               # returns something like "amd64-6.1.63"
-
-          echo VM_KERNEL_VERSION=$kernel_version >> $GITHUB_OUTPUT
+          PROCESSED_TAGS=""
+          for tag in ${TAGS//,/ }; do
+            full_image_tag="$VM_KERNEL_IMAGE-${{ matrix.arch }}:$(echo "$tag" | xargs)"
```

Comment:
suggestion: I spent a couple minutes trying to figure out what these two lines do. I'm sure I could spend more, but perhaps that's an indication of complexity we should either reduce or explain. Can you write a comment explaining what this does? or if you have any ideas to simplify, that'd be welcome, too :sweat_smile:

For reference, things I wasn't sure about:

1. what `${TAGS//,/ }` does
2. why `| xargs`
3. why we need to process these in the first place